cardano-cli/src/Cardano/CLI/EraBased/Options/Query.hs

```diff
@@ -351,8 +352,33 @@ pQueryGetCommitteeStateCmd era envCli = do
   w <- forEraMaybeEon era
   pure
     $ subParser "committee-state"
-    $ Opt.info (QueryCommitteeStateCmd <$> pQueryNoArgCmdArgs w envCli)
+    $ Opt.info (QueryCommitteeMembersStateCmd <$> pQueryCommitteeMembersStateArgs w)
     $ Opt.progDesc "Get the committee state"
+  where
+    pQueryCommitteeMembersStateArgs :: ConwayEraOnwards era -> Parser (QueryCommitteeMembersStateCmdArgs era)
+    pQueryCommitteeMembersStateArgs w = QueryCommitteeMembersStateCmdArgs w
+      <$> pSocketPath envCli
+      <*> pConsensusModeParams
+      <*> pNetworkId envCli
+      <*> many pCommitteeColdVerificationKeyOrHashOrFile
+      <*> many pCommitteeHotKeyOrHashOrFile
+      <*> many pMemberStatus
+      <*> optional pOutputFile
+    pMemberStatus :: Parser MemberStatus
+    pMemberStatus =
+        Opt.option readerMemberStatus $ mconcat
+          [ Opt.long "memberstatus"
+          , Opt.metavar "MEMBER_STATUS"
+          , Opt.help "Member status filter: active/expired/unrecognized"
+          ]
+    readerMemberStatus :: ReadM MemberStatus
+    readerMemberStatus = do
```

Comment:
Is there a standard way to parse an enum without `Read`? Couldn't find anything similar in the code

Reply:
Yes! This is exactly what I was missing. So much better.
Made this change in a new commit.